decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```

Comment:
- [ ] I think it would be better if you use `getter` and `setter` on the `@nameable` instance variable(i.e. `attr_accessor :nameble`) as they these you ensure that you can access and modify the value of a variable, provide a controlled and predictable interface for other objects to interact with the variable, while still maintaining the principles of encapsulation and data hiding.